Add support for multi-line logs

shrishrirang opened the pull request:
- This allows clubbing multiple lines into a single log message, if needed.
- The regex that describes the prefix of each log message is configurable. Example: For simply treating each line as a message, the regex would be ^. To treat lines starting with a time like 21:00:01 as the start of a new log message, regex would be \d\d:\d\d:\d\d

shrishrirang committed:
Add support for multi-line logs

davidebbo commented on TracingLauncher/Program.cs:
```diff
@@ -90,5 +111,73 @@ static int Main(string[] args)
+        {
+            if (_errorStringBuilder.Length > 0)
+            {
+                Trace.TraceInformation(_errorStringBuilder.ToString());
```
Should be `TraceError`, right?

shrishrirang replied:
Yep, fixed.

davidebbo commented on TracingLauncher/Program.cs:
```diff
@@ -2,15 +2,26 @@
 {
     class Program
     {
+        private const int FlushPeriod = 1000;
```
One tick is 100 nanosecond, so that's still a very short amount of time. Did you mean to use milliseconds instead, such that flush is every second?

shrishrirang replied:
Good catch, meant to check for milliseconds. Fixed.

shrishrirang committed:
Improve logging

- Useful in debugging arguments containing special characters like caret. (An unescaped caret on the command line is swallowed by Windows)

shrishrirang committed:
Fix incorrect time check

- Ticks will give nanoseconds, FlushPeriod is in milliseconds. Fixed by using the TotalMilliseconds property instead of Ticks.
